processor/metric/payload.go

```diff
@@ -0,0 +1,300 @@
+package metric
+
+import (
+	"encoding/json"
+	"time"
+
+	"github.com/pkg/errors"
+
+	"github.com/elastic/apm-server/config"
+	m "github.com/elastic/apm-server/model"
+	"github.com/elastic/apm-server/utility"
+	"github.com/elastic/beats/libbeat/beat"
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/libbeat/logp"
+	"github.com/elastic/beats/libbeat/monitoring"
+)
+
+var (
+	transformations = monitoring.NewInt(metricMetrics, "transformations")
+	processorEntry  = common.MapStr{"name": processorName, "event": docType}
+
+	errMissingKey = errors.New("key not found")
+	errNan        = errors.New("not a json number")
+)
+
+type Sample interface {
+	Name() string
+	Value() common.MapStr
+}
+
+func floatFrom(key string, m map[string]interface{}) (float64, error) {
+	v, ok := m[key]
+	if !ok {
+		return 0, errMissingKey
+	}
+	n, ok := v.(json.Number)
+	if !ok {
+		return 0, errNan
+	}
+	return n.Float64()
+}
+
+func stringPtrFrom(key string, m map[string]interface{}) *string {
```

Comment:
same - why not using the `utility/data_fetcher` here? https://github.com/elastic/apm-server/blob/master/utility/data_fetcher.go#L69